src/vs/workbench/services/languageRuntime/common/languageRuntimePlotClient.ts

```diff
@@ -259,26 +276,44 @@ export class PlotClientInstance extends Disposable implements IPositronPlotClien
 		this._register(this._comm);
 	}
 
+	/**
+	 * Get the intrinsic size of the plot, if known.
+	 *
+	 * @returns A promise that resolves to the intrinsic size of the plot, if known.
+	 */
+	public async getIntrinsicSize(): Promise<IntrinsicSize | undefined> {
```

Comment:
It's possible for this to be called quickly in succession since it's invoked during render. To avoid the possibility of stacking intrinsic size requests, we should probably check for an in-flight request here. Maybe use a `DeferredPromise` we can return?

Reply:
Good point, will do. Is there any reason to prefer a `DeferredPromise` over a `Promise` e.g. like:

```typescript
	public getIntrinsicSize(): Promise<IntrinsicSize | undefined> {
		// If there's already an in-flight request, return its response.
		if (this._currentIntrinsicSize) {
			return this._currentIntrinsicSize;
		}
		this._currentIntrinsicSize = this._comm.getIntrinsicSize()
			.then((intrinsicSize) => {
				this._intrinsicSize = intrinsicSize;
				this._receivedIntrinsicSize = true;
				this._didSetIntrinsicSizeEmitter.fire(intrinsicSize);
				return intrinsicSize;
			})
			.finally(() => {
				this._currentIntrinsicSize = undefined;
			});
		return this._currentIntrinsicSize;
	}
```

I've tended to avoid `DeferredPromise` where possible since I tend to forget to try-catch and reject properly.